lib/order.rb

```diff
@@ -0,0 +1,85 @@
+require_relative 'customer.rb'
+require 'csv'
+
+class Order
+
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  SALES_TAX = 0.075
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+
+    raise ArgumentError.new("Invalid fulfillment status. #{fulfillment_status}") unless [:pending, :paid, :processing, :shipped, :complete].include?(fulfillment_status)
```

Comment:
Consider moving this line of code to before any of the instance variables are assigned, since if an argument error is raised, there's not use assigning any instance variables. In this way, this block of code is truly a "guard clause"